### PR TITLE
Add truststore

### LINKED ISSUE #61: Support baseline truststore
Crypto's truststore will be built by nuts-network/nuts-registry when other nodes register certificates. These certificates however are issued by a Root CA which must be trusted too, since it's the trust anchor.

To support this nuts-crypto should have a configuration parameter which can contain the file name of a PEM certificate bundle containing additional entries for the truststore.

### PR DESCRIPTION
**Pull request by reinkrul**
Truststore is used to administer which certificates are trusted, required for registry and nuts-network.

Fixes #61

### CODE REVIEW

File: pkg/cert/truststore.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	pool *x509.CertPool
+	// x509.CertPool doesn't allow you to extract the certificates in it, so we need to keep our own administration.
+	certs []*x509.Certificate
+	file  string
+	// mutex secures concurrent access to AddCertificate() since it might be called concurrently, which is bad since
+	// it does some file manipulation.
+	mutex *sync.Mutex
+}
+
+func (m fileTrustStore) Pool() *x509.CertPool {
```
Comment (woutslakhorst) [inline pkg/cert/truststore.go:57]: I re-enabled the linter so expect some complaining on docs

File: pkg/crypto.go [hunk 2]
```diff
@@ -616,3 +616,34 @@ func (client *Crypto) encryptPlainTextWith(plaintext []byte, key *rsa.PublicKey)
 	}
 	return ciphertext, nil
 }
+
+func serialNumber() (int64, error) {
+	// TODO: Make this implementation safer. This one just hopes for enough entropy.
+	n, err := rand.Int(rand.Reader, big.NewInt(time.Now().UnixNano()*2))
```
Comment (woutslakhorst) [inline pkg/crypto.go]:
```suggestion
	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
```
as used by https://golang.org/src/crypto/tls/generate_cert.go

Comment (reinkrul) [reply pkg/crypto.go]: Good find :+1:

File: pkg/cert/truststore.go (hunk 1 above)
Comment (reinkrul) [reply pkg/cert/truststore.go:57]: Linter does not appear to run? Or it doesn't find anything to complain on (which I find hard to believe).

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**